hotel_reservation/models/hotel_reservation.py

```diff
@@ -0,0 +1,1203 @@
+                checkin_date=checkin_date,
+                checkout_date=checkout_date,
+                duration=False,
+            )
+            duration = duration_vals.get("duration") or 0.0
+            folio_vals = {
+                "date_order": reservation.date_order,
+                "warehouse_id": reservation.warehouse_id.id,
+                "partner_id": reservation.partner_id.id,
+                "pricelist_id": reservation.pricelist_id.id,
+                "partner_invoice_id": reservation.partner_invoice_id.id,
+                "partner_shipping_id": reservation.partner_shipping_id.id,
+                "checkin_date": reservation.checkin,
+                "checkout_date": reservation.checkout,
+                "duration": duration,
+                "reservation_id": reservation.id,
+            }
+            for line in reservation.reservation_line:
+                for r in line.reserve:
+                    folio_lines.append(
+                        (
+                            0,
+                            0,
+                            {
+                                "checkin_date": checkin_date,
+                                "checkout_date": checkout_date,
+                                "product_id": r.product_id and r.product_id.id,
+                                "name": reservation["reservation_no"],
+                                "price_unit": r.list_price,
+                                "product_uom_qty": duration,
+                                "is_reserved": True,
+                            },
+                        )
+                    )
+                    r.write({"status": "occupied", "isroom": False})
+            folio_vals.update({"room_lines": folio_lines})
+            folio = hotel_folio_obj.create(folio_vals)
+            if folio:
+                for rm_line in folio.room_lines:
+                    rm_line.product_id_change()
+            self._cr.execute(
+                "insert into hotel_folio_reservation_rel"
+                "(order_id, invoice_id) values (%s,%s)",
+                (reservation.id, folio.id),
+            )
+            self.state = "done"
+        return True
+
+    def onchange_check_dates(
+        self, checkin_date=False, checkout_date=False, duration=False
+    ):
+        """
+        This method gives the duration between check in checkout if
+        customer will leave only for some hour it would be considers
+        as a whole day. If customer will checkin checkout for more or equal
+        hours, which configured in company as additional hours than it would
+        be consider as full days
+        --------------------------------------------------------------------
+        @param self: object pointer
+        @return: Duration and checkout_date
+        """
+        value = {}
+        configured_addition_hours = 0
+        wc_id = self.warehouse_id
+        whcomp_id = wc_id or wc_id.company_id
```

Comment:
Seems a typo ? should be and 
This is a warehouse or company 

ASFAIK, warehouse is required